examples/speculative_decoding/eagle_utils.py

```diff
@@ -357,7 +367,11 @@ def __call__(self, features: list[dict[str, Any]]) -> dict[str, Any]:
             raise ValueError("No kwargs found in batch features. Offline data required.")
 
         features = [item["kwargs"]["base_model_outputs"] for item in features]
-        max_hs_length = max(item["base_model_hidden_states"].shape[0] for item in features)
+        max_hs_length = (
+            max(item["base_model_hidden_states"].shape[0] for item in features)
+            if self.pad_length is None
+            else self.pad_length
+        )
```

Comment:
nit: just to be consistent to max_length above
```suggestion
       max_hs_length = (
           self.pad_length
            if self.pad_length is not None
            else max(item["base_model_hidden_states"].shape[0] for item in features)
        )
```